datajunction-server/datajunction_server/sql/dag.py

```diff
@@ -80,29 +80,60 @@ def get_dimensions(
     return sorted(list(processed), key=lambda x: x.name)
 
 
-def get_shared_dimensions(
-    metric_nodes: List[Node],
-) -> List[DimensionAttributeOutput]:
+def check_convergence(path1: List[str], path2: List[str]) -> bool:
     """
-    Return a list of dimensions that are common between the nodes.
+    Determines whether two join paths converge before we reach the
```

Comment:
Yeah, I think this is great for now. I can see some folks asking us why two metrics can't be pulled together if their dim paths don't overlap, but I think we can think of that use case later.

Reply:
Yeah, we can add some clearer data modeling examples in the docs, which will also help us clarify the cases we need to support. I'll try to do that at some point